src/Api/IriConverterInterface.php

```diff
@@ -21,6 +21,8 @@
  * Converts item and resources to IRI and vice versa.
  *
  * @author Kévin Dunglas <dunglas@gmail.com>
+ *
+ * @deprecated in favor of ResourceIriConverterInterface to be removed in ApiPlatform 3.0
```

Comment:
I think your concern is legit. We could add a new parameter to the existing method. It would lose a bit of its sense. But really I think you're right about that fact.